industry_benchmarks/utils/data_gathering.py

```diff
@@ -453,14 +474,267 @@ def gather_ligand_scores(
+            raise FileNotFoundError(error_message)
+        else:
+            error_message = f"Can't find cleaned results file: {f_name} in {results_dir} skipping"
+            print(error_message)
+
+
+    return results_dir
+
+def get_transform_name(result: dict, alchemical_network: gufe.AlchemicalNetwork) -> str:
+    """
+    Get the name of this transformation, taking into account that the inputs might have accidentally been deleted.
+    """
+    # grab the gufe key of the chemical systems used in the inputs
+    unit_result = list(result["unit_results"].values())[0]
+    state_a_key = unit_result["inputs"]["stateA"][":gufe-key:"]
+    mapping_key = unit_result["inputs"]["ligandmapping"][":gufe-key:"]
+    # work out which system this is in the alchemical network
+    system_look_up = dict((str(node.key), node) for node in alchemical_network.nodes)
+    mapping_look_up = dict((str(edge.mapping.key), edge.mapping) for edge in alchemical_network.edges)
+    # build the transform
+    if any([isinstance(comp, gufe.ProteinComponent) for comp in system_look_up[state_a_key].components.values()]):
+        phase = "complex"
+    else:
+        phase = "solvent"
+
+    ligmap = mapping_look_up[mapping_key]
+
+    name = f"{phase}_{ligmap.componentA.name}_{ligmap.componentB.name}"
+    return name
+
+def check_network_is_connected(results_data: dict, alchemical_network: gufe.AlchemicalNetwork) -> bool:
+    """Build a network from the results and check the network is connected."""
+    edges = []
+    for transform in alchemical_network.edges:
+        if transform.name in results_data and len(results_data[transform.name]) == 3:
+            edges.append(transform)
+
+    # extract the ligand network and check its connected
+    result_network = gufe.AlchemicalNetwork(edges=edges)
+    ligand_network = extract_ligand_network(result_network)
+    return ligand_network.is_connected()
+
+def get_estimate(result: dict) -> tuple[unit.Quantity, unit.Quantity]:
+    """Extract the DDG and error estimate from this run"""
+    ddg = result["estimate"]["magnitude"] * getattr(unit, result["estimate"]["unit"])
+    uncertainty = result["uncertainty"]["magnitude"] * getattr(unit, result["uncertainty"]["unit"])
+    return ddg, uncertainty
+
+
+def process_results(results_folders: list[pathlib.Path], output_dir: pathlib.Path, alchemical_network: gufe.AlchemicalNetwork) -> dict[str, tuple[unit.Quantity, unit.Quantity]]:
+    """
+    Loop over the results folders extracting the required information and moving it to the output folder.
+
+    Returns
+    -------
+        results: dict[str, tuple[unit.Quantity, unit.Quantity]]
+        The extracted DDG and uncertainty for each transformation repeat
+        > {"solvent_liganda_ligandb_repeat_0": (ddg, uncertainty) ...
+
+    """
+    # workout the expected number of results
+    # assuming 3 repeats of each solvent and complex transformation
+    missing_results = []
```

Comment:
`missing_results` is currently not used. Should we store and print missing results here?

Reply:
Ah now I see you're looking for missing results in the gather_data function. So either remove this empty list here, or store and return missing data already in this function?

Reply:
Now I'm thinking, would it make sense to already collect this information before doing the `network_not_connected` check, so that we could print out the transformations that are missing?